### ISSUE TITLE
Fix the Windows build

### ISSUE THREAD

**Issue by kellabyte** [haywire/haywire#59]
Last time I checked Haywire's multi-threaded support doesn't work on Windows due to a libuv bug that I had trouble tracking down.

Haywire should build on Windows and at least support single threaded mode.

**Comment by Paxxi** [comment]
It builds fine on windows with vs2015 rc, just ran build.bat, plenty of warnings but no errors, didn't try the benchmarking though

**Comment by kellabyte** [comment]
Wow really? I haven't touched it in ages! Any chance you could benchmark single-threaded and multi-threaded mode to make sure it works? I suspect at least multi-threaded mode will be broken.

**Comment by Paxxi** [comment]
first off, http_response_cache.c needs to include time.h otherwise it crashes instantly when built for 64-bit.

wrk is problematic to build on windows so I ignored it and ran it from a virtual linux box instead, command line
```
../bin/wrk/wrk --script ./pipelined_get.lua --latency -d 30s -t 8 -c 32 http://192.168.0.82:8000 -- 64
```
it didn't accept -R as a param for some reason so I removed it.

results are here for single threaded
https://gist.github.com/Paxxi/c6ff33f23c49e57b166c

Trying to run it with 4 threads it dies in haywire\lib\libuv\src\win\stream.c#uv_listen with stream->type set to 0.

**Comment by kellabyte** [comment]
Yeah I suspect the IPC consumer piping I'm doing for multi-threading is a problem on Windows. It did work at one time I believe but then all the sudden stopped working. I suspect a libuv bug but I'm not 100% sure. It could be the Haywire implementation.

Yeah -R is used in wrk2
https://github.com/giltene/wrk2

Wrk2 is good for more correct latency metrics because it accounts for coordinated omission. We should probably include both Wrk for max requests/second benchmarks and Wrk2 for latency benchmarking.

How about removing -R for now from the benchmark script so it doesn't confuse people?

**Comment by kellabyte** [comment]
I also wonder if we are using the wrong pipe name format for Windows to work properly. Can you change it in Haywire and see if it fixes anything?

https://github.com/aspnet/KestrelHttpServer/blob/lodejard/stuff/samples/TcpEcho/Program.cs#L133

**Comment by Paxxi** [comment]
Maybe split benchmark.sh into two, one for wrk and one for wrk2?
I'll see if I have some time during the weekend to test some more and rry that change

**Comment by SajayAntony** [comment]
Hitting an issue compiling on VS 2015 Update 1
Is there a work around for this?

```
"e:\Git\haywire\haywire.sln" (default target) (1) ->
"e:\Git\haywire\haywire_hello_world.vcxproj.metaproj" (default target) (4) ->
"e:\Git\haywire\haywire_shared.vcxproj.metaproj" (default target) (5) ->
"e:\Git\haywire\haywire_shared.vcxproj" (default target) (6) ->
(Link target) ->
  http_request.obj : error LNK2019: unresolved external symbol strndup referenced in function hw_print_request_headers [e:\Git\haywire\haywire_shared.vcxproj]
  e:\Git\haywire\builds\windows\debug\haywire_shared\haywire.dll : fatal error LNK1120: 1 unresolved externals [e:\Git\haywire\haywire_shared.vcxproj]
```

**Comment by nmdguerreiro** [comment]
I think this is fixed by #97

**Comment by SajayAntony** [comment]
@nmdguerreiro  that does fix it :+1:
